app/tests/tests_10/test_external_collection_assets_endpoint.py

```diff
@@ -0,0 +1,373 @@
+import logging
+
+import responses
+
+from django.conf import settings
+from django.test import Client
+from django.test import override_settings
+
+from stac_api.models.collection import CollectionAsset
+
+from tests.tests_10.base_test import TEST_SERVER
+from tests.tests_10.base_test import StacBaseTestCase
+from tests.tests_10.data_factory import Factory
+from tests.tests_10.utils import reverse_version
+from tests.utils import get_auth_headers
+from tests.utils import mock_s3_asset_file
+
+logger = logging.getLogger(__name__)
+
+
+@override_settings(FEATURE_AUTH_ENABLE_APIGW=True)
+class CollectionAssetsExternalAssetEndpointTestCase(StacBaseTestCase):
+
+    @mock_s3_asset_file
+    def setUp(self):  # pylint: disable=invalid-name
+        self.factory = Factory()
+        self.collection = self.factory.create_collection_sample().model
+        self.client = Client(headers=get_auth_headers())
+        self.maxDiff = None  # pylint: disable=invalid-name
+
+    def test_create_collection_asset_with_external_url(self):
+        collection = self.collection
+        collection_asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': settings.EXTERNAL_TEST_ASSET_URL
+        }
+
+        # create the asset, which isn't allowed
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(400, response)
```

Comment:
Test also that the error message is the one that say that external asset are not allowed